app/controllers/course/survey/responses_controller.rb

```diff
@@ -15,11 +15,12 @@ def create
     end
   end
 
-  def show
-    respond_to do |format|
-      format.html
-      format.json
-    end
+  def show; end
+
+  def edit
+    build_missing_answers_and_options
```

Comment:
@kxmbrian Move this method to `response` model instead ? will be easier for testing.

LGTM otherwise.